api/apps/api/test/integration/typeorm-export.repository.integration.e2e-spec.ts

```diff
@@ -0,0 +1,208 @@
+import { ClonePiece, ComponentId, ResourceKind } from '@marxan/cloning/domain';
+import { Connection } from 'typeorm';
+import { v4 } from 'uuid';
+import { ResourceId } from '../../src/modules/clone/export';
+import { ExportEntity } from '../../src/modules/clone/export/adapters/entities/exports.api.entity';
+import { ComponentLocation } from '../../src/modules/clone/export/application/complete-piece.command';
+import { ExportRepository } from '../../src/modules/clone/export/application/export-repository.port';
+import { Export, ExportId } from '../../src/modules/clone/export/domain';
+import { bootstrapApplication } from '../utils/api-application';
+import { FixtureType } from '@marxan/utils/tests/fixture-type';
+import { ExportComponentEntity } from '@marxan-api/modules/clone/export/adapters/entities/export-components.api.entity';
+import { ComponentLocationEntity } from '@marxan-api/modules/clone/export/adapters/entities/component-locations.api.entity';
+
+describe('Typeorm export repository', () => {
+  let fixtures: FixtureType<typeof getFixtures>;
+
+  beforeEach(async () => {
+    fixtures = await getFixtures();
+  }, 20000);
+
+  afterAll(async () => {
+    await fixtures.cleanup();
+  });
+
+  it('should expose methods for getting an export by id and storing exports', async () => {
+    await fixtures.WhenAnExportIsSavedSuccessfully();
+
+    const exportData = await fixtures.WhenReadingTheSavedExportFromRepository();
+
+    await fixtures.ThenExportDataShouldBeOk({
+      exportData,
+      componentsAreCompleted: false,
+    });
+  });
+
+  it('should update an export when a nested piece is completed', async () => {
+    await fixtures.WhenAnExportIsSavedSuccessfully();
+    await fixtures.WhenAComponentIsCompleted();
+    const exportData = await fixtures.WhenReadingTheSavedExportFromRepository();
+
+    await fixtures.ThenExportDataShouldBeOk({
+      exportData,
+      componentsAreCompleted: true,
+    });
+  });
```

Comment:
this may be a bit confusing. Test says `should update an export when a nested piece is complete` so I would rather expect more process/product naming and 'events' what actually happen.

```
GivenExportWasRequested();
WhenLastPieceWasCompleted();
ThenExportIsSaved();
ThenExportFinalizedEventIsSent();  // don't remember the name, sorry
```

Reply:
As this is an integration test for our custom repository, this test should not know about events. Also, the second step is just one completed piece, not every one / the last one